src/Npgsql/NpgsqlConnector.cs

```diff
@@ -957,6 +958,13 @@ internal ValueTask<IBackendMessage> ReadMessage(bool async, DataRowLoadingMode d
                     }
                     throw;
                 }
+                catch (NpgsqlException)
+                {
+                    // An ErrorResponse isn't followed by ReadyForQuery
+                    if (error != null)
+                        throw error;
```

Comment:
Never do this. Use [`ExceptionDispatchInfo`](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.exceptionservices.exceptiondispatchinfo) to prevent stack trace replacement:

```csharp
ExceptionDispatchInfo().Throw(error);
```

Reply:
Thank you for reviewing.

I can't build `ExceptionDispatchInfo().Throw(error);`. So I use `ExceptionDispatchInfo.Capture(error).Throw();`.

Do you mean the latter?